### PR TITLE
MINOR: Gradle and Gradle plugins versions are extracted into 'gradle/dependencies.gradle' file

### PR DESCRIPTION
**Pull request by dejan2609**
### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dejan2609** [comment]
Prologue: https://github.com/apache/kafka/pull/6436#pullrequestreview-231478142

@ijuma, @ewencp: please review and let me know do you prefer `spotbugs_plugin` over `spotbugsPlugin` (so I can change that)

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -78,16 +82,24 @@ versions += [
...
   slf4j: "1.7.25",
   snappy: "1.1.7.2",
+  spotbugs: "3.1.12",
+  spotbugsPlugin: "1.6.9",
+  spotlessPlugin: "3.23.0",
   zkclient: "0.11",
   zookeeper: "3.4.13",
-  zstd: "1.3.8-1"
+  zstd: "1.3.8-1",
+  versionsPlugin: "0.21.0"
```
Comment (ijuma) [inline gradle/dependencies.gradle]: I think we should call this `gradleVersionsPlugin`.

File: gradle/dependencies.gradle [hunk 2]
```diff
@@ -78,16 +82,24 @@ versions += [
   mavenArtifact: "3.6.0",
   metrics: "2.2.0",
   mockito: "2.23.4",
+  owaspPlugin: "4.0.2",
```
Comment (ijuma) [inline gradle/dependencies.gradle]: owaspDepCheckPlugin?

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. Just a couple of minor comments.

File: gradle/dependencies.gradle (hunk 1 above)
Comment (dejan2609) [reply gradle/dependencies.gradle]: Done, name changed.

File: gradle/dependencies.gradle (hunk 2 above)
Comment (dejan2609) [reply gradle/dependencies.gradle]: Done, name changed.

**Comment by dejan2609** [comment]
Minor variable name changes (suggested by @ijuma) are applied.

**Comment by dejan2609** [comment]
retest this please

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Comment by ijuma** [comment]
The PR builder for Java 11 is currently failing due to an Apache Infra problem. There's no behaviour change in this PR so will go ahead and merge to trunk.

**Pull request merged**